src/features/Product/components/Comments.js

```diff
@@ -0,0 +1,118 @@
+import Textarea from "components/Textarea";
+import Button from "components/Button";
+
+const Comments = ({ itemId }) => {
+  const [comments, setComments] = useState([]);
+  const [editingId, setEditingId] = useState(null);
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  const handleLoad = useCallback(async () => {
+    const { list } = await getComments(itemId);
+    setComments(list);
+  }, [itemId]);
+
+  const handleSelect = (id, option) => {
+    if (option === "edit") {
+      setEditingId(id);
+    }
+  };
+
+  const handleCancel = () => {
+    setEditingId(null);
+  };
+
+  const handleTextareaChange = (event) => {
+    if (event.target.value !== "") {
+      setIsDisabled(false);
+      return;
+    }
+
+    setIsDisabled(true);
+  };
+
+  useEffect(() => {
+    handleLoad();
+  }, [handleLoad]);
+
+  return (
+    <section>
+      {comments.length ? (
+        <ul className={styles.comments}>
+          {comments.map(({ content, id, createdAt, writer }) => {
+            if (id === editingId) {
+              return (
+                <li key={id} className={styles.comment}>
+                  <Textarea
+                    value={content}
+                    className={styles.textarea}
+                    onChange={handleTextareaChange}
+                  />
+                  <div className={styles.editContainer}>
+                    <AuthorInfo
+                      className={styles.authorInfo}
+                      nickname={writer.nickname}
+                      image={writer.image}
+                      date={timeAgo(createdAt)}
+                    />
+                    <div className={styles.buttonContainer}>
+                      <button
+                        type="button"
+                        className={styles.cancelButton}
+                        onClick={handleCancel}
+                      >
+                        취소
+                      </button>
+                      <Button
+                        className={styles.editButton}
+                        type="submit"
+                        disabled={isDisabled}
+                      >
+                        수정완료
+                      </Button>
+                    </div>
+                  </div>
+                </li>
+              );
+            }
+
+            return (
+              <li key={id} className={styles.comment}>
+                <div className={styles.contentContainer}>
+                  <p className={styles.content}>{content}</p>
+                  <Dropdown
+                    className={styles.dropdown}
+                    onSelect={(option) => handleSelect(id, option)}
+                  />
+                </div>
+                <AuthorInfo
+                  className={styles.authorInfo}
+                  nickname={writer.nickname}
+                  image={writer.image}
+                  date={timeAgo(createdAt)}
+                />
+              </li>
+            );
+          })}
```

Comment:
P2:
너무 길어서 파악하기가 어렵네요~
컴포넌트로 분리하면 가독성에 더 좋을 것 같습니다.

```suggestion
          {comments.map(({ content, id, createdAt, writer }) => {
            return id === editingId ? <EditingComment /> : <Comment />
           })}
```

Reply:
위의 `EditingComment`에서만 사용되는 로직도 해당 컴포넌트 내부로 옮겨주시면 최고 🙌